onnxruntime_extensions/tools/pre_post_processing/step.py

```diff
@@ -166,27 +166,37 @@ class Debug(Step):
                 to rename so it's more consistent.
     """
 
-    def __init__(self, num_inputs: int = 1, name: Optional[str] = None):
+    def __init__(self, num_inputs: int = 4, name: Optional[str] = None, custom_func: Optional[callable] = None):
```

Comment:
I think num_inputs should be always equal to the real num of inputs. I want to set it as -1 by default.
Why '4' here, usally, mode of graph wouldn't has inputs more than 4, and may we don't want to add more than 4 outputs if there is one. 
3 inputs are quite normally in NLP and 1 input in CV, '4' is always working good.

Sorry, will add some comments for `custom_func `.
This function usally works for saving the graph. Comparing with add outputs for debug in the whole graph, a small "debug.onnx" might be better and easier to debug.

Reply:
'4' seems arbitrary and maybe it's cleaner/more consistent to simply require the user of the Debug class to specify the number of inputs if it's more than 1, instead of attempting to over-allocate and reducing the number of inputs later. The over-allocation also potentially breaks logic in Step.connect where it validates an IoMapEntry, as input_names won't necessarily match the real number of inputs.

Regarding outputs, I would have expected we want to provide debug outputs for all the inputs as we don't know which are important. If the user finds that noisy they can use an IoMapEntry to reduce the inputs to the Debug step.

Regarding custom_func, I'm wary of polluting the Step API for a temporary feature that might only be useful when someone is developing the Step. They could hardcode writing out the onnx model if that's the main usage, or create a pipeline with the Step last and run it against an empty model. 

That seems different to the Debug step adding outputs which is more about seeing the actual values when running the full model than checking the structure of the graph for the step.

Reply:
Indeed.  num_inputs=4 will lead to a error when built connections.

Yeah, I agree, custom_func is more using for developers.